models/Element/Recyclebin/Item.php

```diff
@@ -329,6 +326,67 @@ protected function doRecursiveRestore(Element\ElementInterface $element)
         }
     }
 
+    /**
+     * @param Element\ElementInterface $data
+     *
+     * @return mixed
+     */
+    public function marshalData($data)
```

Comment:
please have a look at @dpfaffenbauer comment about the custom filter and incorporate his idea if possible